src/app/[locale]/clientSearchDataset.ts

```diff
@@ -0,0 +1,25 @@
+import {SearchOptions, SearchResult} from '@/lib/dataset-fetcher';
+
+interface SearchDatasets {
+  licenses: string[];
+  publishers: string[];
+}
+
+// only use this function for client components
```

Comment:
Minor: uppercase to 'Only' (for consistency - the sentence underneath starts with 'For')